docs/transparency.rst

```diff
@@ -0,0 +1,421 @@
+Most users don't have to worry much about what the methods mean. Though it's good to understand
+that the "opaque" and "stochastic" methods produce opaque fragments, and by default have ``depth_write=True``.
+The renderer sorts these objects front-to back to avoid overdraw (for performance).
+
+In contrast, the "composite" and "weighted" methods result in semi-transparent fragments,
+and by default have ``depth_write=False``. The renderer sorts these object back-to-front to
+improve the chance of correct blending. Note that the 'auto' mode is an exception to this rule.
+
+
+Alpha methods in detail
+-----------------------
+
+**Alpha method 'opaque'** represents no transparency. The fragment color
+overwrites the value in the output texture. A very common method in render engines.
+
+**Alpha method 'composite'** represents alpha compositing: a common method in render
+engines in which objects are combined on a per-fragment basis. The object's
+fragment color and the current color in the output texture are blended using a
+mathematical formula. There are several common compositing configurations, the
+most-used being the "over operator" (also known as normal blending). When alpha
+compositing is used, the result will depend on the order in which the objects are
+rendered.
+
+**Alpha method 'stochastic'** represents stochastic transparency. The alpha
+represents the chance of a fragment being visible (i.e. not discarded). Visible
+fragments are opaque. This blend method is less common, but has interesting properties.
+Although the result has a somewhat noisy appearance, it handles transparency perfectly,
+capable of rendering multiple layers of transparent objects, and correctly handling
+objects that have a mix of opaque and transparent fragments.
+
+**Alpha method 'weighted'** represents (variants of) weighted blended order
+independent transparency. The order of objects does not matter for the
+end-result. One use-case being order independent transparency (OIT).
+The order-independent property is advantageous in some use-cases, but produces
+unfavourable results in others. It's use extends beyond transparency though, and
+can also be used for e.g. image stiching.
+
+
+Alpha config
+------------
+
+The ``material.alpha_config`` is a dictionary that fully describes how the combining based on alpha occurs.
+This dictionary has at least two keys: the 'method' and 'mode'. It has additional keys for the options
+available for the used method. The different presets represent common combinations of these options.
+
+Most users just set ``material.alpha_mode`` which implicitly sets
+``material.alpha_config``. In advanced/special cases, users can set the
+``material.alpha_config`` directly to take full control over all available
+options. In this case the 'mode' field and ``material.alpha_mode`` become "custom".
+
+
+Render queue
+------------
+
+The ``material.render_queue`` is an integer that represents the group that the renderer uses to sort objects.
```

Comment:
Maybe also mention here that it is an integer and not an Enum because that would prevent custom assignment? I can already picture the PRs/issues about this.